### DISCUSSION TITLE
Add Sound

### DISCUSSION THREAD

**Discussion by Kite54** [google/model-viewer#1931] [Q&A]
Hi… how can I add sound to the scene?

There should be a sound parameter with a synchronised valid url – but it doesn't work for me?

 <model-viewer 
src=".."
sound="…"
</model-viewer>

Thanks!

### Description

Please provide a detailed description of the bug, how to reproduce it, and the
expected behavior.  Always include a code snippet, screenshot, any errors
reported in the console, and the model to help us understand and fix the
problem.

#### Live Demo

https://glitch.com/edit/#!/model-viewer

### Version

- model-viewer: vX.X.X

### Browser Affected

- [ ] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS

- [ ] Android
- [ ] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
Can you give me some context for your question? There is no `sound` attribute in the model-viewer element, though you can use standard web techniques to add sound to your page. Are you referring to Scene Viewer documentation? I don't keep such close tabs on their project, so if so, please add a link to their documentation that you're trying to follow.

**Comment by Kite54** [comment]
Hi elalish…

this is the Link where i got the Information from.
https://developers.google.com/ar/develop/java/scene-viewer

I just want to play some music when the ar start… in best case that it is syschrnonized… i tried to put the sound in the gab file… but of course that is not supported;)

this is the Link to my test Project:

https://ksih.design/clickandbuilds/WordPress/MyBlog/Downloads/app4.8/

It would be grateful if you have a tipp for me! I'm more a designer than a coder;)

Thanks!

**Comment by Kite54** [comment]
Im mean the glb file of course… not gab file!;)

**Comment by elalish** [comment]
Okay, so these are query parameters, which means you put them inside of the `src` attribute URL, which we then pass on to scene-viewer. So try something like `src="my-model-url?sound=my-sound-url` and you'd add additional parameters by separating them with `&`.

**Comment by Kite54** [comment]
Hi Elalish, thanks for the support…

this works fine on OS… but not on mobile device like android or iOS.
There must be a solution for both mobile systems like this!?

https://vr.google.com/scene-viewer?file=https://storage.googleapis.com/ar-answers-in-search-models/static/EmperorPenguin/model.glb&title=Pinguine&sound=https://storage.googleapis.com/ar-answers-in-search-models/static/EmperorPenguin/Penguin_Emperor_Unisex_Adult.ogg

(The link works only on android;)

Thanks!

**Comment by elalish** [comment]
Well, you asked about a Scene Viewer feature, and Scene Viewer is an Android app (part of AGSA), so no, it is not a cross-platform solution. If you want it to work everywhere, you should instead be looking for web solutions like the `<audio>` tag. This will work naturally in our default WebXR AR mode too (on Chrome Android), but not on Scene Viewer or Quick Look. Until iOS supports WebXR, you'll always have to make a special .usdz or .reality file that also includes the sound to get the same experience on iOS.

**Comment by Kite54** [comment]
Hi elalish, thanks for the answer… but i don't get it…

For iOS – i tried to embed the sound in the usdz file using aero or glb converter, also the real file is not supported by (my!?) iOS maybe you have an idea how to embed sound in the usdz file?

For Android – i also tried the audio tag with autoplay - but it is also do not work on my android with the newest chrome version?

Thanks!

**Reply by dthery** [reply]
@Kite54 have you put a file in OGG format?
I did a quick test on android, it works.

**Comment by Kite54** [comment]
Hi… thanks for the support. actually i don’t remember what i tried and what not… i only know that it doesn’t work for me…

Can you please give me the code you used? I would like to give it another try!;) 

Thanks!

**Comment by AndresSalomon1990** [comment]
Following this topic. I'm also interested in adding sound to the AR scene.

**Comment by PixelPartner** [comment]
@Kite54 The race went into a new lap.

Apple introduced to play audio with their `preliminary_` prefixed USD schemes for iOS13.1+.
It's a bigger effort to create an interactive scene with RealityComposer.

With iOS18.1, they now support `SpatialAudio` as specified in this [UsdAudio-Proposal](https://openusd.org/docs/UsdAudio-Proposal.html) (not officially announced, but I tried successfully) 

```usda
    def SpatialAudio "Audio"
    {
        double3 xformOp:translate     = (0.0, 0.0, 0.0)
        uniform token[] xformOpOrder  = ["xformOp:translate"]
        uniform token playbackMode   = "onceFromStart"
        uniform asset filePath       = @myAudio.mp3@
        uniform token auralMode      = "spatial"
        uniform timeCode startTime   =  0.0 # in timeCode units
        uniform double mediaOffset   =  0.0 # in seconds
    }
```

It's now much easier to add audio to an asset. Place the audio file inside the USDZ zip archive and use `usdcat` and a text editor to add the above snippet.
